iris/src/iris/pipeline/tutor_suggestion_pipeline.py

```
@@ -0,0 +1,354 @@
+        self,
+        state: AgentPipelineExecutionState[
+            CommunicationTutorSuggestionPipelineExecutionDTO, TutorSuggestionVariant
+        ],
+    ) -> list[Callable]:
+        allow_lecture_tools = should_allow_lecture_tool(state.db, state.dto.course.id)
+        allow_faq_tool = should_allow_faq_tool(state.db, state.dto.course.id)
+        is_programming_exercise = state.dto.programming_exercise is not None
+        is_text_exercise = state.dto.text_exercise is not None
+
+        if not hasattr(state, "lecture_content_storage"):
+            setattr(state, "lecture_content_storage", {})
```

Ruff (0.12.2)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)